explainability-service/src/test/java/org/kie/trustyai/service/endpoints/data/DataEndpointTest.java

```diff
@@ -439,4 +444,118 @@ void uploadTagThatUsesProtectedName() {
                         Dataframe.TRUSTYAI_INTERNAL_TAG_PREFIX,
                         Dataframe.TRUSTYAI_INTERNAL_TAG_PREFIX)));
     }
+
+    private void postTest(Object payload, int statusCode, List<String> checkMsgs) {
+        ValidatableResponse r = given()
+                .contentType(ContentType.JSON)
+                .body(payload)
+                .when().post("/upload")
+                .then();
+
+        System.out.println(r.extract().body().asString());
+        for (String checkMsg : checkMsgs) {
+            r.statusCode(statusCode).body(containsString(checkMsg));
+        }
+    }
+
+    @Test
+    void uploadDataAndGroundTruth() {
+        int[] testInputRows = new int[] { 1, 5, 250 };
+        int[] testInputCols = new int[] { 1, 4 };
+        int[] testOutputCols = new int[] { 1, 2 };
+        String[] testDatatypes = new String[] { "INT64", "INT32", "FP32", "FP64", "BOOL" };
+
+        // sorry for the quad loop
```

Comment:
@RobGeada could you create a ticket to optimise this, and replace the comment with the issue link? Thanks.